File: src/main/java/com/wlyscraft/slimefunitemexpansion/machines/DeathGenerator.java
package com.wlyscraft.slimefunitemexpansion.machines;

import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AGenerator;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import javax.annotation.Nonnull;

public class DeathGenerator extends AGenerator {
    public DeathGenerator(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, recipeType, recipe);
    }

    @Nonnull
    @Override
    public ItemStack getProgressBar() {
        return new ItemStack(Material.BONE_BLOCK);
    }

    @Override
    protected void registerDefaultFuelTypes() {
        this.registerFuel(new MachineFuel(1, new ItemStack(Material.ROTTEN_FLESH)));
        this.registerFuel(new MachineFuel(1, new ItemStack(Material.BONE)));
        this.registerFuel(new MachineFuel(3, new ItemStack(Material.BONE_BLOCK)));
        this.registerFuel(new MachineFuel(45, new ItemStack(Material.WITHER_SKELETON_SKULL)));
    }
}

File: src/main/kotlin/io/github/itsflicker/slimefunitemexpansion/items/machines/DeathGenerator.kt
package io.github.itsflicker.slimefunitemexpansion.items.machines

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AGenerator
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class DeathGenerator(itemGroup: ItemGroup, item: SlimefunItemStack, recipeType: RecipeType, recipe: Array<ItemStack?>) :
    AGenerator(itemGroup, item, recipeType, recipe) {

    override fun getProgressBar(): ItemStack {
        return ItemStack(Material.BONE_BLOCK)
    }

    override fun registerDefaultFuelTypes() {
        registerFuel(MachineFuel(1, ItemStack(Material.ROTTEN_FLESH)))
        registerFuel(MachineFuel(1, ItemStack(Material.BONE)))
        registerFuel(MachineFuel(3, ItemStack(Material.BONE_BLOCK)))
        registerFuel(MachineFuel(45, ItemStack(Material.WITHER_SKELETON_SKULL)))
    }
}